src/__tests__/test-utils/use-mutation.util.ts

```diff
@@ -1,19 +1,16 @@
 import { act, renderHook } from '@testing-library/react-hooks'
+import { MutateFunction } from 'react-query'
 
-export default async function executeMutation(callback: any, ...input: any[]) {
-  let mutateToTest: any
-  await act(async () => {
-    const renderHookResult = renderHook(callback)
-    const { result, waitForNextUpdate } = renderHookResult
-    await waitForNextUpdate()
-    const [mutate] = result.current as any
-    mutateToTest = mutate
-  })
+export default async function executeMutation<TResult>(
```

Comment:
Making this generic is what caused most of the changes in this review.  The cleanup here is actually a drop in replacement for the old version.